metals/src/main/scala/scala/meta/internal/builds/BSPErrorHandler.scala

```diff
@@ -0,0 +1,124 @@
+package scala.meta.internal.builds
+
+import java.nio.file.Files
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.control.NonFatal
+
+import scala.meta.internal.metals.ClientCommands
+import scala.meta.internal.metals.Directories
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.clients.language.MetalsLanguageClient
+import scala.meta.io.AbsolutePath
+
+import org.eclipse.{lsp4j => l}
+
+class BSPErrorHandler(
+    languageClient: MetalsLanguageClient,
+    workspaceFolder: AbsolutePath,
+    restartBspServer: () => Future[Boolean],
+)(implicit context: ExecutionContext) {
```

Comment:
```suggestion
class BSPErrorHandler(
    languageClient: MetalsLanguageClient,
    workspaceFolder: AbsolutePath,
    restartBspServer: () => Future[Boolean],
    currentSession: () => BspSession
)(implicit context: ExecutionContext) {
```
and move the logic that checks `session.main.isBloop || session.main.isScalaCLI` to this class? To keep everything here?